directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java

```diff
@@ -120,12 +121,20 @@ public static RequestException forFormPostMissingParameters(List<String> missing
   public static RequestException forPollEventNotFound(String eventName) {
     assert !StringUtils.isEmpty( eventName );
 
-    return new RequestException( "No method registered for poll event '" + eventName + "'" );
+    return new RequestException( "No method registered for poll event '" + htmlEncode(eventName) + "'" );
   }
 
   public static RequestException forSourceNotFound( String sourceName) {
     assert !StringUtils.isEmpty(sourceName);
     
-    return new RequestException( "Unable to find source for '" + sourceName + "'");
+    return new RequestException( "Unable to find source for '" + htmlEncode(sourceName) + "'");
+  }
+
+  private static String htmlEncode(String value) {
+    if (value == null) {
+      return value;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') com.softwarementors.extjs.djn.router.processor.RequestException.htmlEncode(String) may return null, but is declared @Nonnull [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.s/coding_rules#rule_key=findbugs%3ANP_NONNULL_RETURN_VIOLATION)